tests/strategies/filter/test_crop_filter.py

```diff
@@ -34,14 +35,16 @@ def test_crop_filter(static_files: "Path") -> None:
     image_parser: "IParseStrategy" = ImageDataParseStrategy(image_config)
 
     # Run "pipeline"
-    session = image_parser.initialize()
+    session = {}
     session.update(crop_filter.initialize(session))
-    session.update(crop_filter.get(session))
+    session.update(image_parser.initialize(session))
     session.update(image_parser.get(session))
+    session.update(crop_filter.get(session))
+
+    cache = DataCache()
+    image_key = session["image_key"]
+    data = cache.get(image_key)
+
+    assert data.shape == (400, 700, 3)
 
-    assert session.get("cropped", False)
-    assert session.get("format", SupportedFormat).value == "JPEG"
-    assert (
-        session.get("content", b"")
-        == (static_files / "sample_700_400.jpeg").read_bytes()
-    )
+    del cache[image_key]
```

Comment:
Put this in a `try/finally` block with the `del` line in `finally` and the `data =  ...` and `assert ...` lines are in the `try` block.

Reply:
Yes, maybe we should that, but by default the cache expire within 14 days.

A better cleanup strategy would be to either ask the conftest to set up a new data cache for this testing session that is flushed and deleted when the tests have run.

Another strategy, which would be useful outside testing, is to create a separate Session class (could e.g. be a subclass of AttrDict) with a custom __del__() method that evict all items in the cached that are tagged with the session id `str(id(session))`. That way we can dump temporary data for this session in the cache and be sure it will be cleaned up when the session goes out of scope.

Reply:
Added issue #114 for an even better solution, that should be easy to implement. I would prefer that in front of poluting the tests with a lot of `try/finally`...

Reply:
I like the idea about a general data cache for all tests, however, this breaks the test compartmentalization a bit. Another solution there is to create a context fixture for the data cache that returns a new data cache and flushes it when exiting.

Concerning a separate `Session` class that flushes the data cache that seems overkill and not the job of session. That should be the job of the data cache or similar.

Reply:
Could you please either put this in a `try/finally` block here - or add a context fixture to do so?

Reply:
done